### DISCUSSION TITLE
Can't connect to hostname on LAN

### DISCUSSION THREAD

**Discussion by Kestrelio** [mcguirepr89/BirdNET-Pi#205] [Q&A] [answered]
This may be an issue with my router... Not sure. I can SSH via the IP, and my browser can pull up a blank page via the IP, but nothing seems to connect to the hostname I set. If I ping the hostname from outside my RPi, I get nothing, but if I ping it from inside the RPi it appears to work. So I know I've got the right hostname, but it doesn't appear to be broadcasting on my network. Thoughts?

**Comment by ehpersonal38** [comment]
I would suggest searching for a way to set a custom "/etc/hosts" entry on your specific router, as your router doesn't appear to be doing the DNS/DHCP things it needs to be doing automatically. Also make sure your client devices (computer, phone) have your router's IP as their primary DNS server. 

Hopefully someone with more knowledge about this can chime in!

**Reply by Kestrelio** [reply]
Yeah this is probably the issue. I have a custom DNS on my primary computer and I'm locked out of my modem/router admin panel. So it'll probably take a total factory reset of the hardware for me to get back into it, yadda yadda.

In the meantime, I don't mind connecting over IP, except that I can't seem to actually pull up the Birdnet homepage. Is there a port I should be pointing to? 80 is open, and presumably what the server is broadcasting over, but again I just come up with a blank page.

**Comment by mcguirepr89** [comment, marked as answer]
Hi, @Kestrelio!!
First of all, thank you so very much for installing BirdNET-Pi!!! I hope you find it worth this extra bit of trouble!

To respond to your supposition, it is almost definitely your router not allowing "multicast DNS" or "mDNS" announcements on the network. The BirdNET-Pi will send an mDNS annoucement that provides its IP address and corresponding avahi-alias (your hostname.local) to all hosts on the network so that the router doesn't need to keep track of the DNS entry.

Since your router is blocking this announcement, no other devices on the network know about the BirdNET-Pi by its avahi-alias. To address this, you have a few options!

To try and get the hostname working:
1. Find the settings in your router that will allow for mDNS, enable it, then reboot the BirdNET-Pi

If you can't get your router to allow mDNS announcements, you can have the installation use its IP address instead.
To switch the installation over to using the IP address instead of a hostname:
1. SSH into the installation.
2. Edit the file `$HOME/BirdNET-Pi/birdnet.conf` so that your "`BIRDNETPI_URL=`" line contains your IP address with the HTTP protocol, for example:
   ```
   BIRDNETPI_URL=http://192.168.42.1
   ```
3. Run ```sudo update_caddyfile.sh``` and then you should be able to reach your installation using its IP address instead of its hostname. After this step, you should be able to see your installation by going to "http://192.168.42.1" (following the example IP).
4. Finally, you'll want to go into your router and look for DHCP settings that will allow you to reserve the IP that was issued to the BirdNET-Pi so that its lease is "static," meaning the DHCP server on your router will always give the BirdNET-Pi the same IP address.

Please let me know if you have any questions or encounter any trouble and I will be more than happy to help!!!
My best regards,
Patrick

**Reply by Kestrelio** [reply]
That did the trick! Thank you. Looking forward to using this.

**Reply by mcguirepr89** [reply]
I'm really glad -- may I ask, did you get mDNS working on the router, or did you switch the installation over to using the IP?

**Reply by Kestrelio** [reply]
I just switched to IP. Long story short, whatever the admin password is for my router got lost, so I'll need to factory reset at some point. IP works fine though.

Great work on this, I gotta say. Been running it all morning on my RPi400 with an old Logitech headset. The mic isn't great but the system is performing well and producing accurate results. I'll be interested to see where you take this next.

Out of curiosity, does the system do anything when there's no internet? Is it capturing audio samples for later upload and processing? I was going to fashion my RPi400 into a lightweight laptop for an upcoming backpacking trip. Would really like to collect some bird audio then.

**Reply by mcguirepr89** [reply]
> Out of curiosity, does the system do anything when there's no internet? Is it capturing audio samples for later upload and processing?

Yep! It does all analysis locally, so even without internet it will just keep working. I'm putting together a pre-installed image that has an automatic HotSpot for fully offline data collection -- you just connect you phone/laptop to the hotspot, then view the installation from the Pi's network. 

The only offline quirk is that there is no RTC in the Pi, so it if loses power (or is booted without network connectivity) it won't have accurate time stamps. Otherwise, losing internet doesn't stop any data collection nor extraction :)

**Reply by ryankrage77** [reply]
> there is no RTC in the Pi

There are RTC clocks that connect over I2C available. They're not exactly plug 'n' play though, for example see [this pihut blog post](https://thepihut.com/blogs/raspberry-pi-tutorials/17209332-adding-a-real-time-clock-to-your-raspberry-pi).
